src/main/java/org/apache/aurora/scheduler/mesos/MesosTaskFactory.java

```diff
@@ -214,8 +217,19 @@ public TaskInfo createFrom(IAssignedTask task, Offer offer) throws SchedulerExce
           taskBuilder.setExecutor(execBuilder.build());
         } else {
           LOG.warn("Running Docker-based task without an executor.");
+
           taskBuilder.setContainer(getDockerContainerInfo(dockerContainer, Optional.absent()))
-              .setCommand(CommandInfo.newBuilder().setShell(false));
+                     .setCommand(CommandInfo.newBuilder().setShell(false));
+
+          if (config.isSetKillPolicy()) {
+            DurationInfo.Builder durationBuilder;
+            durationBuilder = DurationInfo.newBuilder()
+                    .setNanoseconds(config.getKillPolicy().getGracePeriod() * 1000000000);
+
+            KillPolicy.Builder killPolicyBuilder = KillPolicy.newBuilder()
```

Comment:
build() here (create the KillPolicy) and then assign the end result?
Inlining would work too.
Creating a builder is not very common.